t/chaos/killetcd/killetcd.go

```diff
@@ -0,0 +1,190 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package killetcd
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	"github.com/gavv/httpexpect"
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/api/v1/pod"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/apache/apisix/t/chaos/utils"
+)
+
+var (
+	bandwidthBefore float64
+	durationBefore  float64
+	bpsBefore       float64
+	bandwidthAfter  float64
+	durationAfter   float64
+	bpsAfter        float64
+)
+
+func getEtcdKillChaos() *v1alpha1.PodChaos {
+	return &v1alpha1.PodChaos{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "kill-etcd",
+			Namespace: metav1.NamespaceDefault,
+		},
+		Spec: v1alpha1.PodChaosSpec{
+			Selector: v1alpha1.SelectorSpec{
+				LabelSelectors: map[string]string{"app.kubernetes.io/instance": "etcd"},
+			},
+			Action: v1alpha1.PodKillAction,
+			Mode:   v1alpha1.AllPodMode,
+			Scheduler: &v1alpha1.SchedulerSpec{
+				Cron: "@every 10m",
+			},
+		},
+	}
+}
+
+var _ = ginkgo.Describe("Test Get Success When Etcd Got Killed", func() {
+	e := httpexpect.New(ginkgo.GinkgoT(), utils.Host)
+	eSilent := httpexpect.WithConfig(httpexpect.Config{
```

Comment:
A recommended best practice is putting all your initialization logics into `ginkgo.BeforeEach`.